src/Catalyst.KBucket.Tests/CollectionTest.cs

```diff
@@ -0,0 +1,205 @@
+            bucket.Add(new Contact("a"));
+            Assert.AreEqual(6, bucket.Count);
+        }
+
+        [TestMethod]
+        public void Clear()
+        {
+            var bucket = new KBucket<Contact>();
+            Assert.AreEqual(0, bucket.Count);
+
+            bucket.Add(new Contact("a"));
+            bucket.Add(new Contact("b"));
+            bucket.Add(new Contact("c"));
+            Assert.AreEqual(3, bucket.Count);
+
+            bucket.Clear();
+            Assert.AreEqual(0, bucket.Count);
+        }
+
+        [TestMethod]
+        public void Remove()
+        {
+            var bucket = new KBucket<Contact>();
+            Assert.AreEqual(0, bucket.Count);
+
+            bucket.Add(new Contact("a"));
+            bucket.Add(new Contact("b"));
+            bucket.Add(new Contact("c"));
+            Assert.AreEqual(3, bucket.Count);
+
+            bucket.Remove(new Contact("b"));
+            Assert.AreEqual(2, bucket.Count);
+
+            Assert.IsTrue(bucket.Contains(new Contact("a")));
+            Assert.IsFalse(bucket.Contains(new Contact("b")));
+            Assert.IsTrue(bucket.Contains(new Contact("c")));
+        }
+
+        [TestMethod]
+        public void CopyTo()
+        {
+            var bucket = new KBucket<Contact>();
+            Assert.AreEqual(0, bucket.Count);
+
+            bucket.Add(new Contact("a"));
+            bucket.Add(new Contact("b"));
+            bucket.Add(new Contact("c"));
+            Assert.AreEqual(3, bucket.Count);
+
+            var array = new Contact[bucket.Count + 2];
+            bucket.CopyTo(array, 1);
+            Assert.IsNull(array[0]);
+            Assert.IsNotNull(array[1]);
+            Assert.IsNotNull(array[2]);
+            Assert.IsNotNull(array[3]);
+            Assert.IsNull(array[4]);
+        }
+
+        [TestMethod]
+        public void Enumerate()
+        {
+            var bucket = new KBucket<Contact>();
+            var nContacts = bucket.ContactsPerBucket + 1;
+            for (var i = 0; i < nContacts; ++i)
+            {
+                bucket.Add(new Contact(i));
+            }
+
+            Assert.AreEqual(nContacts, bucket.Count);
+
+            var n = 0;
+            
+            foreach (var _ in bucket)
+            {
+                ++n;
+            }
+            
+            Assert.AreEqual(nContacts, n);
+        }
+
+        [TestMethod]
+        public void CanBeModified() { Assert.IsFalse(new KBucket<Contact>().IsReadOnly); }
+
+        [TestMethod]
+        public async Task ThreadSafe()
+        {
+            var bucket = new KBucket<Contact>();
+            const int nContacts = 1000;
+            const int nTasks = 100;
+            var tasks = new Task[nTasks];
+
+            for (var i = 0; i < nTasks; ++i)
+            {
+                var start = i;
+                tasks[i] = Task.Run(() => AddTask(bucket, start, nContacts));
+            }
+
+            await Task.WhenAll(tasks);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)